tsdb/engine/wal/wal.go

```diff
@@ -455,7 +458,7 @@ func (l *Log) nextMetaFile() error {
 			return err
 		}
 
-		id = int(n)
+		id = int(n) + 1
```

Comment:
both are necessary since changing a shared variable in goroutines should be protected by a mutex.